File: PythonWhiteLibrary/WhiteLibrary.py
import clr
clr.AddReferenceToFileAndPath('CSWhiteLibrary.dll') #include full path to Dll if required
from CSWhiteLibrary import Keywords
WHITE_LIB = Keywords()


class WhiteLibrary(object):

    def launch_application(self, sut):
        '''
        Launch windows application
        | Arguments | Usage | (M)andatory / (O)ptional |
        | sut | application under testing | M |
        '''
        WHITE_LIB.launch_application(sut)

    def attach_window(self, window):
        '''
        Attach to window in application
        | Arguments | Usage | (M)andatory / (O)ptional |
        | window | window to attach | M |
        '''
        WHITE_LIB.attach_window(window)

    def close_application(self):
        '''
        Close application
        | No arguments |
        '''
        WHITE_LIB.close_application()

    def set_logging_level(self, level):
        '''
        Set log level
        | Arguments | Usage | (M)andatory / (O)ptional |
        | level | log level (info / debug / warn) | M |
        '''
        WHITE_LIB.set_log_level(level)

    def input_text_to_textbox(self, locator, text):
        '''
        Write text to textbox
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        | text | inserted string | M |
        '''
        WHITE_LIB.input_text_textbox(locator, text)

    def verify_text_in_textbox(self, locator, expected):
        '''
        Verify text in textbox
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        | actual | expected text | M |
        '''
        actual = WHITE_LIB.verify_text_textbox(locator)
        verify_value(expected, actual)

    def verify_label(self, locator, expected):
        '''
        Verify text in label
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        | actual | expected text | M |
        '''
        actual = WHITE_LIB.verify_label(locator)
        verify_value(expected, actual)

    def select_combobox_value(self, locator, value):
        '''
        Select combobox value
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        | value | selected value | M |
        '''
        WHITE_LIB.select_combobox_value(locator, value)

    def select_combobox_index(self, locator, index):
        '''
        Select combobox index
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        | index | combobox index | M |
        '''
        WHITE_LIB.select_combobox_index(locator, int(index))

    def verify_combobox_item(self, locator, expected):
        '''
        Verify combobox selected value
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        | actual | expected value | M |
        '''
        actual = WHITE_LIB.verify_combobox_item(locator)
        verify_value(expected, actual)

    def verify_button(self, locator, expected):
        '''
        Verify text in button
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        | actual | expected text | M |
        '''
        actual = WHITE_LIB.verify_button(locator)
        verify_value(expected, actual)

    def click_button(self, locator):
        '''
        Click button
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        '''
        WHITE_LIB.click_button(locator)

    def verify_menu(self, locator, expected):
        '''
        Verify menu item exists
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        | expected | expected text in value | M |
        '''
        actual = WHITE_LIB.verify_menu(locator)
        verify_value(expected, actual)

    def click_menu_button(self, locator):
        '''
        Click menu button
        | Arguments | Usage | (M)andatory / (O)ptional |
        | locator | element id | M |
        '''
        WHITE_LIB.click_menu_button(locator)

    def select_modal_window(self, locator):
        '''
        Select modal window by title
        | Arguments | Usage | (M)andatory / (O)ptional |
        | Title | Modal window title | M |
        '''
        WHITE_LIB.select_modal_window(locator)

def verify_value(expected, actual):
    if expected != actual:
        raise AssertionError("Correct value not found: %s != %s" % (expected, actual))

File: WhiteLibrary/WhiteLibrary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TestStack.White;
using TestStack.White.Factory;
using TestStack.White.ScreenObjects.Services;
using TestStack.White.ScreenObjects.Sessions;
using TestStack.White.UIItems;
using TestStack.White.UIItems.MenuItems;
using TestStack.White.UIItems.ListBoxItems;
using TestStack.White.UIItems.Finders;
using TestStack.White.UIItems.WindowItems;
using WhiteLibrary;


namespace CSWhiteLibrary
{
    public class Keywords : WhiteFW
    {
        private Application app;

        public void launch_application(string sut)
        {
            this.app = Application.Launch(sut);
        }

        public void attach_window(string window)
        {
            this.window = app.GetWindow(window);
        }

        public void set_log_level(string level)
        {
            set_logging(level);
        }

        public void close_application()
        {
            app.Close();
        }

        public void input_text_textbox(string locator, string mytext)
        {
            TextBox textBox = getTextBox(locator);
            textBox.Text = mytext;
        }

        public void set_slider(string locator, Double myvalue)
        {
            Slider myslider = getSlider(locator);
            myslider.Value = myvalue;
        }

        public string verify_text_textbox(string locator)
        {
            TextBox textBox = getTextBox(locator);
            return textBox.Text;
        }
     
        public Double verify_slider(string locator)
        {
            Slider mySlider = getSlider(locator);
            return mySlider.Value;
        }

        public Double verify_progressbar(string locator)
        {
            ProgressBar myProgressBar = getProgressBar(locator);
            return myProgressBar.Value;
        }

        public string verify_label(string locator)
        {
            Label label = getLabel(locator);
            return label.Text;
        }
        
        public void select_combobox_value(string locator, string value)
        {
            ComboBox comboBox = getComboBox(locator);
            comboBox.Select(value);
        }
    
        public void select_combobox_index(string locator, int index)
        {
            ComboBox comboBox = getComboBox(locator);
            comboBox.Select(index);
        }

        public void select_listbox_value(string locator, string value)
        {
            ComboBox listBox = getComboBox(locator);
           // ListItem listItem = getListItem(value);
           // listBox.Select(getListItem(value).ToString());
        }

        public string verify_listbox_value(string locator, string value)
        {
            ListBox listBox = getListBox(locator);
            return listBox.SelectedItemText;
        }

        public string verify_combobox_item(string locator)
        {
            ComboBox comboBox = getComboBox(locator);
            return comboBox.EditableText;
        }

        public string verify_button(string locator)
        {
            Button button = getButton(locator);
            return button.Text;
        }

        public void click_button(string locator)
        {
            Button button = getButton(locator);
            button.Click();
        }

        public void select_radio_button(string locator)
        {
            RadioButton radio_button = getRadioButton(locator);
            radio_button.Select();
        }
        
        public Boolean verify_radio_button(string locator)
        {
            RadioButton radio_button = getRadioButton(locator);
            return radio_button.IsSelected;
        }

        public Boolean verify_check_box(string locator)
        {
            CheckBox check_box = getCheckBox(locator);
            return check_box.IsSelected;
        }

        public void toggle_check_box(string locator)
        {
            CheckBox check_box = getCheckBox(locator);
            check_box.Toggle();
        }

        public string verify_menu(string locator)
        {
            Menu menu = getMenu(locator);
            return menu.Name;
        }

        public void click_menu_button(string locator)
        {
            Menu menu = getMenu(locator);
            menu.Click();
        }

        public void select_modal_window(string locator)
        {
            List<Window> modalWindows = window.ModalWindows();
            this.window = window.ModalWindow(locator);
        }
    }
}